Sources/StreamChat/WebSocketClient/EventMiddlewares/ChannelReadUpdaterMiddleware.swift

```diff
@@ -59,12 +61,19 @@ struct ChannelReadUpdaterMiddleware: EventMiddleware {
     private func incrementUnreadCountIfNeeded(
         for cid: ChannelId,
         message: MessagePayload,
-        session: DatabaseSession
+        session: DatabaseSession,
+        notificationCenter: EventNotificationCenter
     ) {
         guard let currentUser = session.currentUser else {
             return log.error("Current user is missing", subsystems: .webSocket)
         }
 
+        // If the message exists in the database before processing the current batch of events, it means it was
+        // already processed and we don't have to increase the unread count
+        guard notificationCenter.newMessageIds.contains(message.id) else {
```

Comment:
@polqf Why don't we simply fetch the message from the `DatabaseSession` to check if it already exists? Why do we need the notification centre? I'm 100% following it 🤔

Reply:
It does not work as there is another middleware saving the message to the db just before the unread count is processed... And I don't want to rely on the order of the middlewares to do this